master/internal/configpolicy/task_config_policy_intg_test.go

```diff
@@ -0,0 +1,84 @@
+package configpolicy
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+func TestPriorityAllowed(t *testing.T) {
+	require.NoError(t, etc.SetRootPath(db.RootFromDB))
+	pgDB, cleanup := db.MustResolveNewPostgresDatabase(t)
+	defer cleanup()
+	db.MustMigrateTestPostgres(t, pgDB, db.MigrationsFromDB)
+
+	// When no constraints are present, any priority is allowed.
+	ok, err := PriorityAllowed(1, model.NTSCType, 0, true)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	wkspLimit := 50
+	w := addWorkspacePriorityLimit(t, pgDB, wkspLimit)
+
+	// Priority is outside workspace limit.
+	ok, err = PriorityAllowed(w.ID, model.NTSCType, wkspLimit-1, true)
+	require.NoError(t, err)
+	require.False(t, ok)
+
+	globalLimit := 42
+	addGlobalPriorityLimit(t, pgDB, globalLimit)
+
+	// Priority is within global limit.
+	ok, err = PriorityAllowed(w.ID, model.NTSCType, wkspLimit-1, true)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	// Priority is outside global limit.
+	ok, err = PriorityAllowed(w.ID+1, model.NTSCType, globalLimit-1, true)
+	require.NoError(t, err)
+	require.False(t, ok)
+}
+
+func addWorkspacePriorityLimit(t *testing.T, pgDB *db.PgDB, limit int) model.Workspace {
```

Comment:
small nit but I kinda prefer setting `user := db.RequireMockUser(t, pgDB)` in `TestPriorityAllowed` instead and passing that down to the helpers instead of retrieving the val in both helpers. WDYT?

Reply:
Hmm ... good point. In this case, `user` isn't needed outside of the helper; I chose to set it within the helper to make the helper more self-sufficient and reduce the number of required arguments. In other words, the user is created for the sole purpose of setting the task config policies - so why should another function be responsible for creating the user?

It would eliminate one database call, but it doesn't seem to be a particularly expensive call or anything.

Reply:
I'm happy to make the change though!